Magic.cs

```diff
@@ -166,7 +166,7 @@ private static void onAnalyze(object sender, AnalyzeEventArgs e)
                     Log.debug("Player learnt spell: " + spell);
                     farmer.learnSpell(spell, 0, true);
                     //Game1.drawObjectDialogue(Mod.instance.Helper.Translation.Get("spell.learn", new { spellName = Mod.instance.Helper.Translation.Get("spell." + spell + ".name") }));
-                    Game1.addHUDMessage(new HUDMessage(Mod.instance.Helper.Translation.Get("spell.learn", new { spellName = Mod.instance.Helper.Translation.Get("spell." + spell + ".name") })));
+                    Game1.addHUDMessage(new HUDMessage(Mod.instance.Helper.Translation.Get("spell.learn", new { spellName = SpellBook.get(spell).getTranslation("spell." + spell + ".name") })));
```

Comment:
Also, here, instead of using `SpellBook.get(spell)` you can just use `spell` directly.

Reply:
Did you extend string somehow? Because `List<string> spellsLearnt = new List<string>();` is the list `spell` comes from.

Reply:
Whoops, you're right. I misread.